lib/hooks/index.js

```diff
@@ -97,25 +67,51 @@ function findModuleVersion(request, parent, load) {
     if (mainScriptDir.indexOf(paths[i]) === 0) {
       // This is unsafe on windows if request contains a namespace
       // TODO: convert path separators in request
-      var pjson = load(path.join(paths[i], request, 'package.json'));
-      return pjson.version;
+      return path.join(paths[i], request);
     }
   }
   return null;
 }
 
+/**
+ * Determines the version of the module located at `modulePath`.
+ *
+ * @param {?string} modulePath The absolute path to the root directory of the
+ *    module being loaded. This may be null if we are loading an internal module
+ *    such as http.
+ */
+function findModuleVersion(modulePath, load) {
+  return modulePath ? load(path.join(modulePath, 'package.json')).version :
```

Comment:
The contract of `findModulePath` is that it will either return a valid path to a module or null based on the use of `Module#_resolveFilename` and `Module#_resolveLookupPaths`. Since the argument passed into `findModuleVersion` is the return value of a call to `findModulePath`, the load should succeed. We can add additional checks or fold this functionality into the `findModulePath` logic but in the current form I don't think this can fail.